MdeModulePkg/Bus/Pci/XhciDxe/Xhci.c

```diff
@@ -2077,7 +2077,11 @@ XhcDriverBindingStart (
 
   XhcSetBiosOwnership (Xhc);
 
-  XhcResetHC (Xhc, XHC_RESET_TIMEOUT);
+  Status = XhcResetHC (Xhc, XHC_RESET_TIMEOUT);
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "%a: failed to reset HC\n", __func__));
+    return Status;

```

Comment:
What is the expected functionality of the XHCI controller if it returns here?

The event crated in XhcCreateUsbHc will still be installed and receiving callbacks.

Reply:
Thanks. I've changed the `return Status` line to `goto FREE_POOL`.